src/atomate2/common/jobs/pheasy.py

```diff
@@ -0,0 +1,475 @@
+    max_atoms: int,
+    force_90_degrees: bool,
+    force_diagonal: bool,
+) -> list[list[float]]:
+    """
+    Determine supercell size with given min_length and max_length.
+
+    Parameters
+    ----------
+    structure: Structure Object
+        Input structure that will be used to determine supercell
+    min_length: float
+        minimum length of cell in Angstrom
+    max_length: float
+        maximum length of cell in Angstrom
+    prefer_90_degrees: bool
+        if True, the algorithm will try to find a cell with 90 degree angles first
+    allow_orthorhombic: bool
+        if True, orthorhombic supercells are allowed
+    **kwargs:
+        Additional parameters that can be set.
+    """
+    transformation = CubicSupercellTransformation(
+        min_length=min_length,
+        max_atoms=max_atoms,
+        force_90_degrees=force_90_degrees,
+        force_diagonal=force_diagonal,
+        angle_tolerance=1e-2,
+        allow_orthorhombic=False,
+    )
+    transformation.apply_transformation(structure=structure)
+    return transformation.transformation_matrix.transpose().tolist()
+
+
+@job(data=[Structure])
+def generate_phonon_displacements(
+    structure: Structure,
+    supercell_matrix: np.array,
+    displacement: float,
+    num_displaced_supercells: int,
+    cal_anhar_fcs: bool,
+    displacement_anhar: float,
+    num_disp_anhar: int,
+    fcs_cutoff_radius: list[int],
+    sym_reduce: bool,
+    symprec: float,
+    use_symmetrized_structure: str | None,
+    kpath_scheme: str,
+    code: str,
+) -> list[Structure]:
+    """
+
+    Generate small-distance perturbed structures with phonopy based on two ways.
+
+    (we will directly use the pheasy to generate the supercell in the near future)
+    1. finite-displacment method (one displaced atom) when the displacement number
+    is less than 3. 2. random-displacement method (all-displaced atoms) when the
+    displacement number is more than 3.
+
+    Parameters
+    ----------
+    structure: Structure object
+        Fully optimized input structure for phonon run
+    supercell_matrix: np.array
+        array to describe supercell matrix
+    displacement: float
+        displacement in Angstrom (default: 0.01)
+    num_displaced_supercells: int
+        number of displaced supercells defined by users
+    sym_reduce: bool
+        if True, symmetry will be used to generate displacements
+    symprec: float
+        precision to determine symmetry
+    use_symmetrized_structure: str or None
+        primitive, conventional or None
+    kpath_scheme: str
+        scheme to generate kpath
+    code:
+        code to perform the computations
+
+    """
+    warnings.warn(
+        "Initial magnetic moments will not be considered for the determination "
+        "of the symmetry of the structure and thus will be removed now.",
+        stacklevel=1,
+    )
+    cell = get_phonopy_structure(
+        structure.remove_site_property(property_name="magmom")
```

Comment:
ideally, we export the common parts from the phonons and pheasy jobs into a method and reuse it? This would then also make it easier when we change anything here in the future in either of the jobs in pheasy.py or phonons.py